File: tests/fixtures/csharp-lexical-syntax/host/src/CSharpCohort/BillingModel.cs
using System.Collections.Generic;
using System.Linq;

namespace CSharpCohort;

// decision:0001 keeps the source-only C# boundary explicit.
public sealed record Invoice(string Id, int Amount);

public abstract record BillingOutcome
{
    public sealed record Accepted(Invoice Invoice) : BillingOutcome;
    public sealed record Rejected(string Reason) : BillingOutcome;
}

public static class BillingModel
{
    public static string Label(this Invoice invoice) => $"{invoice.Id}:{invoice.Amount}";

    public static int Total(Invoice invoice) => invoice.Amount;

    public static int Total(IReadOnlyList<Invoice> invoices) =>
        invoices.Sum(invoice => invoice.Amount);
}

File: tests/fixtures/kotlin-lexical-syntax/host/src/main/kotlin/cohort/BillingModel.kt
package cohort

// decision:0001 keeps the source-only Kotlin boundary explicit.
data class Invoice(val id: String, val amount: Int)

sealed interface BillingOutcome {
    data class Accepted(val invoice: Invoice) : BillingOutcome
    data class Rejected(val reason: String) : BillingOutcome
}

fun Invoice.label(): String = "$id:$amount"

fun total(invoice: Invoice): Int = invoice.amount

fun total(invoices: List<Invoice>): Int = invoices.sumOf { it.amount }
